src/coreclr/src/jit/importer.cpp

```diff
@@ -18596,8 +18598,7 @@ void Compiler::impInlineRecordArgInfo(InlineInfo*   pInlineInfo,
         INDEBUG(curArgVal->AsLclVar()->gtLclILoffs = argNum;)
     }
 
-    if ((curArgVal->OperKind() & GTK_CONST) ||
-        ((curArgVal->gtOper == GT_ADDR) && (curArgVal->AsOp()->gtOp1->gtOper == GT_LCL_VAR)))
+    if ((curArgVal->OperKind() & GTK_CONST) || isAddressInLocal)
```

Comment:
There's some code between these 2 changes that doesn't appear in GitHub:
```C++
    if (curArgVal->gtFlags & GTF_ALL_EFFECT)
    {
        inlCurArgInfo->argHasGlobRef = (curArgVal->gtFlags & GTF_GLOB_REF) != 0;
        inlCurArgInfo->argHasSideEff = (curArgVal->gtFlags & (GTF_ALL_EFFECT & ~GTF_GLOB_REF)) != 0;
    }
```
It looks to me that in some cases this will unnecessarily set `argHasGlobRef` when the arg is a local address. Not sure what impact that has on CQ, if any.

Reply:
I saw your comment about that over on dotnet/coreclr#27630. But I didn't see any global ref flag on the test case here, at least when inlining.

I can look into bypassing that for invariant args.

Reply:
Interesting, here I see it in M1 on FIELD and ADDR:
```
STMT00002 (IL 0x00F...  ???)
               [000012] I-C-G-------              *  CALL      ref    System.Int32.ToString (exactContextHnd=0x02F7E975)
               [000011] ----G------- this in ecx  \--*  ADDR      byref 
               [000010] ----G-------                 \--*  FIELD     int    i
               [000009] ------------                    \--*  ADDR      byref 
               [000008] ------------                       \--*  LCL_VAR   int    V01 loc0         
```